### PR TITLE
Add MRCPP package

### PR DESCRIPTION
**Pull request by robertodr**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by robertodr** [comment]
@conda-forge/help-c-cpp how am I supposed to get different MPI variants working? `conda_build_config.yml` looks correct, but I keep getting `Error: Failed to render jinja template in /home/conda/staged-recipes-copy/recipes/mrcpp/meta.yaml:
'mpi' is undefined`

File: conda_build_config.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 mpi:
```
Comment (scopatz) [inline conda_build_config.yaml:1]: Please don't edit any files outside of your recipe directory.  Instead add a new `conda_build_config.yaml` in your recipe dir for that recipe

File: recipes/mrcpp/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  license: LGPL-3.0-or-later
+  license_url: https://opensource.org/licenses/LGPL-3.0
+  license_file: LICENSE
+  license_family: LGPL
+  summary: "MultiResolution Computational Program Package"
+
+extra:
+  recipe-maintainers:
+    - robertodr
+    - bjorgve
```
Comment (scopatz) [inline recipes/mrcpp/meta.yaml:109]: @bjorgve are you willing to be a maintainer?

Comment (bjorgve) [reply recipes/mrcpp/meta.yaml:109]: @scopatz hi yes I am.

File: recipes/mrcpp/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+  run_exports:
+    - {{ pin_subpackage('mrcpp', max_pin='x.x') }}
+  run_exports:
+    - {{ name }} * {{ mpi_prefix }}_*
+
+requirements:
+  build:
+    - cmake >=3.11
+    - {{ compiler('cxx') }}
+    - {{ pin_compatible('eigen', max_pin='x.x') }}
```
Comment (scopatz) [inline recipes/mrcpp/meta.yaml:58]: eigen should probably be in host

Comment (robertodr) [reply recipes/mrcpp/meta.yaml]: Done

File: recipes/mrcpp/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build }}
+  skip: true  # [win]
+  binary_relocation: true
+  string: "{{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}"
+  run_exports:
+    - {{ pin_subpackage('mrcpp', max_pin='x.x') }}
+  run_exports:
```
Comment (scopatz) [inline recipes/mrcpp/meta.yaml:51]: I think there should only be one `run_exports` key

Comment (robertodr) [reply recipes/mrcpp/meta.yaml:51]: You mean, unify under one section:
```yaml
  run_exports:
    - {{ pin_subpackage('mrcpp', max_pin='x.x') }}
    - {{ name }} * {{ mpi_prefix }}_*
```
or is the second line redundant, given that `string` is present?

Comment (robertodr) [reply recipes/mrcpp/meta.yaml:51]: Looking at the HDF5 `meta.yaml` (from which I adapted this one) I guess it should be:
```yaml
run_exports:
   - {{ pin_subpackage('mrcpp', max_pin='x.x') }} {{ build_pin }}
```

Comment (scopatz) [reply recipes/mrcpp/meta.yaml:51]: I mean unify under one section

Comment (scopatz) [reply recipes/mrcpp/meta.yaml:51]: I don't think it needs the build_pin

Comment (robertodr) [reply recipes/mrcpp/meta.yaml]: Thanks :+1: I've consolidated `run_exports` now

**Comment by scopatz** [comment]
Thanks!

**Comment by robertodr** [comment]
I see that the Travis build failed and the mrcpp-feedstock repo was not created. Is there anything I should do to fix this?

**Pull request merged**